### PR TITLE
Set minimumWidth of WLabel according to the content's maximum possible width

### PR DESCRIPTION
**Pull request by daschuer**
This fixes the wiggling off the knob position during adjusting the value.

A regression caused by the switch between value and label.

### CODE REVIEW

File: src/widget/weffectparameternamebase.cpp [hunk 1]
```diff
@@ -66,6 +78,11 @@ void WEffectParameterNameBase::parameterUpdated() {
         m_text = kNoEffectString;
         setBaseTooltip(tr("No effect loaded."));
     }
+    optimumWidth = math_max(
+            optimumWidth,
+            metrics.size(0, m_text).width());
+    setMinimumWidth(optimumWidth + 6);
```
Comment (ronso0) [inline src/widget/weffectparameternamebase.cpp:84]: Why +6 ?
I tested with Tango & LateNight and it looks good without it, even with crammed effect slots, +6 only shifts the 'elide' threshold a bit. Is that it?

**Review by ronso0 — CHANGES_REQUESTED** [review]
LGTM, thanks for the fixup!

Just a question, and please remove the debug output (--amend)

**Comment by ronso0** [comment]
This screws up the label alignment in Tango and LateNight, though by looking at the Shade knobs xml I discovered how to simplify the Tango/LateNight knob templates.
fix: d639dad6fc

**Comment by daschuer** [comment]
Re-implementing the sizeHint() Function did finally the trick. Please test again.

File: src/widget/weffectparameternamebase.cpp [hunk 2]
```diff
@@ -66,6 +82,10 @@ void WEffectParameterNameBase::parameterUpdated() {
         m_text = kNoEffectString;
         setBaseTooltip(tr("No effect loaded."));
     }
+    optimumWidth = math_max(
+            optimumWidth,
+            metrics.size(0, m_text).width());
+    m_widthHint = optimumWidth;
```
Comment (ronso0) [inline src/widget/weffectparameternamebase.cpp:88]: let's rename `optimumWidth` to `valueWidth` and 
```suggestion
    m_widthHint = math_max(
            valueWidth,
            metrics.size(0, m_text).width());
```
and remove the debug ouput below

**Comment by ronso0** [comment]
I guess the minimumSizeHint doesn't consider the padding.
If I remove the padding the label is static
```
           padding-left: 3px;
           padding-right: 3px;
```
https://github.com/mixxxdj/mixxx/pull/11194/files#diff-e1924755bc957e48039a69bcc6c083ed94ae7d4570ef7f317595d370fd3ba5fbR26-R27

However, this is what happens:
* start Mixxx with Shade, fx section is hidden, Graphic EQ is unit1, slot2
* show fx
Screenshot_2023-02-02_23-30-18
* start to drag a parameter label, or reload the effect:
the entire knob container shrinks in width:
Screenshot_2023-02-02_23-30-27

File: src/widget/wlabel.cpp [hunk 3]
```diff
@@ -95,11 +96,8 @@ void WLabel::setText(const QString& text) {
     if (m_elideMode != Qt::ElideNone) {
         QFontMetrics metrics(font());
         // Measure the text for label width
-        // it turns out, that "-2" is required to make the text actually fit
-        // (Tested on Ubuntu Trusty)
-        // TODO(lp#:1434865): Fix elide width calculation for cases where
-        // this text is next to an expanding widget.
-        QString elidedText = metrics.elidedText(m_longText, m_elideMode, width() - 2);
+        m_widthHint = metrics.size(0, m_longText).width() + frameWidth();
+        QString elidedText = metrics.elidedText(m_longText, m_elideMode, width() - frameWidth());
```
Comment (ronso0) [inline src/widget/wlabel.cpp:100]:
```suggestion
        m_widthHint = metrics.size(0, m_longText).width() + 2 * frameWidth();
        QString elidedText = metrics.elidedText(m_longText, m_elideMode, width() - 2 * frameWidth());
```

Comment (ronso0) [reply src/widget/wlabel.cpp:100]: I think that was it.
At least now the size doesn't change anymore...

Though I think we should try to figure why the label shrinks when parameters are swapped.

**Review by ronso0 — CHANGES_REQUESTED** [review]

File: src/widget/weffectparameternamebase.cpp (hunk 2 above)
Comment (ronso0) [inline src/widget/weffectparameternamebase.cpp:88]: let's rename `optimumWidth` to `valueWidth` and 
```suggestion
    int optimumWidth = math_max(
            valueWidth,
            metrics.size(0, m_text).width());
    m_widthHint = optimumWidth + 2 * frameWidth();
```
and remove the debug ouput below

**Review by ronso0 — CHANGES_REQUESTED** [review]

File: src/widget/weffectparameternamebase.cpp (hunk 2 above)
Comment (ronso0) [reply src/widget/weffectparameternamebase.cpp:88]: with this the labels are okay, besides the fact that they are initially (before swapping parameters) expanded compared to 2.4

**Comment by daschuer** [comment]
Thank you for the good testing. For me it is working now. Also after parameter swapping.

**Comment by ronso0** [comment]
Yes, it works now.

It's just that something 'broke' the initial view in Shade, compared to 2.4:
initial view:
Screenshot_2023-02-06_01-18-38
swapped parameters:
Screenshot_2023-02-06_01-18-57

I this actually looks better, but users may be confused / disappointed by the condensed view after the swap.
Though tbh I don't care too much about it. Since the other skins still work great this is ready.
If you're motivated to debug this we can wait, otherwise I'll merge it. Just let me know how you prefer it.

**Review by ronso0 — APPROVED** [review]

**Comment by ronso0** [comment]
@daschuer Ignore this "good"/harmless regression, or investiagte once more?

**Comment by daschuer** [comment]
Let's just merge this. It is IMHO good enough.

**Pull request merged**